iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/toggle/annotation/FeatureGateAspect.java

```diff
@@ -0,0 +1,30 @@
+package com.linkedin.openhouse.internal.catalog.toggle.annotation;
+
+import com.linkedin.openhouse.internal.catalog.toggle.ResourceGatedByToggledOnFeatureException;
+import com.linkedin.openhouse.internal.catalog.toggle.TableFeatureToggle;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Aspect
+@Component
+@Slf4j
+public class FeatureGateAspect {
+
+  @Autowired private TableFeatureToggle tableFeatureToggle;
+
+  @Before("@annotation(featureGate)")
+  public void checkIcebergFeatureFlag(JoinPoint joinPoint, IcebergFeatureGate featureGate) {
+    TableIdentifier tableIdentifier = (TableIdentifier) joinPoint.getArgs()[0];
```

Comment:
is there an assumption that first parameter will always be tableidentifier? if so can we add it to the documentation?

Reply:
I did some generalization of this method, like not limiting `TableIdentifier` to be the first parameter. 
But the constraints of method being annotated having to have a `TableIdentifier` object still exists. Thus, I am alway naming the annotation with `Iceberg` in it which should answer the question here: https://github.com/linkedin/openhouse/pull/174#discussion_r1744618336